orchestration/dap_orchestration/solids.py

```diff
@@ -241,8 +242,59 @@ def write_outfiles(context: AbstractComputeExecutionContext, fan_in_results: lis
     NOTE: The fan_in_results param allows to introduce a fan-in dependency from upstream transformation
     solids, but is ignored by this solid.
     """
+    tsv_directory = context.solid_config["output_dir"]
     context.resources.outfiles_writer.run(
-        context.solid_config["output_dir"],
+        tsv_directory,
         context.resources.refresh_directory,
         fan_in_results
     )
+
+    return tuple((fan_in_results, tsv_directory))
+
+
+@solid(
+    required_resource_keys={"refresh_directory", "gcs"},
+    config_schema={
+        "upload_dir": String,
+    }
+)
+def upload_to_gcs(
+        context: AbstractComputeExecutionContext,
+        inputs: tuple
+    ) -> None:
+    """
+    This solid will take upload the tsvs created from write_outfiles to the provided gs:// bucket. The script
+    will check that the file exists before uploading it and will error if it does not exist.
+
+    NOTE: The fan_in_results param allows to introduce a fan-in dependency from the upstream write_outfiles
+    solid and is used to iterate through TSV uploads for the surveys being refreshed.
+    # todo: handle having to upload with a different name as well? - sample.tsv + sample_MMDDYYYY.tsv
+    # todo: add firecloud flag to tsv script for sample outfile generation
+        # alternately we can hardcode entity:sample_id into the tsv script for the sample table
+        # the same could be done to handle the duplicate renamed file
+    """
+    fan_in_results = inputs[0]
+    tsv_directory = inputs[1]
+    for survey in fan_in_results:
+        storage_client = context.resources.gcs
+        tsv_dir = f"{tsv_directory}/tsv_output"
+        upload_dir = context.solid_config["upload_dir"]
+
+        tsvBucketWithPrefix = parse_gs_path(tsv_dir)
+        uploadBucketWithPrefix = parse_gs_path(upload_dir)
```

Comment:
⛏️ For consistency, I'd say rename this 
```suggestion
        destBucketWithPrefix = parse_gs_path(upload_dir)
```